app-hw-37/src/Store/auth.ts

```diff
@@ -22,13 +29,38 @@ export const authSlice = createSlice({
 		setUserPassword: (state, action: PayloadAction<string>) => {
 			state.userPassword = action.payload
 		},
-		handleAuth: (state) => {
-			state.isAuthorised = true
+		handleAuth: (state, action: PayloadAction<User>) => {
+			const user = state.users.find((user) => user.login === action.payload.login && user.pass === action.payload.pass)
+			if (user) {
+				state.userLogin = action.payload.login
+				state.userPassword = action.payload.pass
+				state.isAuthorised = true
+			}
 		},
 		handleLogOut: (state) => {
 			state.isAuthorised = false
+		},
+		addUser: (state, action: PayloadAction<User>) => {
+			state.users.push(action.payload)
 		}
-	}
+	},
+	extraReducers(builder) {
+		builder.addCase(authRedirection.fulfilled, (state, action: PayloadAction<boolean>) => {
+			if (action.payload) {
+				console.log('success')
+				redirect('success')
+			} else {
+				console.log('fail')
+				redirect('fail')
+			}
+		})
+	},
+})
+
+export const authRedirection = createAsyncThunk('auth/redirect', async () => {
+	const result = await store.getState()
```

Comment:
in your code you have race condition
you don't know who will be first `store.getState()` or `handleAuth`